source/extensions/quic_listeners/quiche/platform/quic_mem_slice_impl.h

```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+// NOLINT(namespace-envoy)
+
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include <iostream>
+#include <memory>
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/assert.h"
+
+#include "quiche/quic/core/quic_buffer_allocator.h"
+
+namespace quic {
+
+class QuicMemSliceImpl {
```

Comment:
I expected this to inherit from QuicMemSlice per Envoy convention. Any thoughts on why this is different?

Reply:
Hi Joshua, this is actually quic convention to platform-ize code in different code base.
https://quiche.googlesource.com/quiche has the core quiche code + platform API's, but it doesn't platform impl's.

There are on-going  effort to implements the API's in the quiche directory: https://quiche.googlesource.com/quiche/+/refs/heads/master/quic/platform/api. And this PR is one of them.

Reply:
Yeah, see the disclaimer at the comment at top of file :P

Reply:
SG. Can you add a comment above the class declaration something like this:
```
// Implements the interface required by
// https://quiche.googlesource.com/quiche/+/refs/heads/master/quic/platform/api/quic_mem_slice.h
// Note that this is a link-time plug-in rather than using inheritance.
```
That will help casual readers understand what this is plugging into, which is otherwise hard to grep for.